server/asset/eth/coiner.go

```diff
@@ -284,3 +286,151 @@ func (c *swapCoin) Value() uint64 {
 func (c *swapCoin) FeeRate() uint64 {
 	return c.gasPrice
 }
+
+var _ asset.FundingCoin = (*amountCoin)(nil)
+
+type amountCoin struct {
+	backend *Backend
+	cID     *AmountCoinID
+}
+
+func newAmountCoin(backend *Backend, coinID []byte) (*amountCoin, error) {
+	cID, err := DecodeCoinID(coinID)
+	if err != nil {
+		return nil, err
+	}
+	aCoinID, ok := cID.(*AmountCoinID)
+	if !ok {
+		return nil, errors.New("coin ID not an amount")
+	}
+	if _, err := accountHasBalance(backend, &aCoinID.Address, aCoinID.Amount); err != nil {
+		return nil, err
+	}
+	return &amountCoin{
+		backend: backend,
+		cID:     aCoinID,
+	}, nil
```

Comment:
This uses the same nonce we get from the client. Is it important to ensure it is unique for the server?

Reply:
Makes sense to me.